sdk/core/azure-core/inc/azure/core/datetime.hpp

```diff
@@ -113,107 +243,77 @@ namespace Azure { namespace Core {
     {
-      return ToString(DateFormat::Iso8601, fractionFormat);
+      return GetString(DateFormat::Rfc3339, fractionFormat);
     };
 
-    /// Get the integral time value.
-    IntervalType ToInterval() const { return m_interval; }
+    DateTime& operator+=(Duration const& value)
+    {
+      m_since1601 += value;
+      return *this;
+    }
+
+    DateTime& operator-=(Duration const& value)
+    {
+      m_since1601 += value;
+      return *this;
+    }
 
     /// Subtract an interval from @DateTime.
-    DateTime operator-(IntervalType value) const { return DateTime(m_interval - value); }
+    DateTime operator-(Duration const& value) const { return DateTime(m_since1601 - value); }
 
     /// Add an interval to @DateTime.
-    DateTime operator+(IntervalType value) const { return DateTime(m_interval + value); }
+    DateTime operator+(Duration const& value) const { return DateTime(m_since1601 + value); }
+
+    /// Get duration between two instances of @DateTime.
+    Duration operator-(DateTime const& other) const { return m_since1601 - other.m_since1601; }
 
     /// Compare two instances of @DateTime for equality.
-    bool operator==(DateTime dt) const { return m_interval == dt.m_interval; }
+    constexpr bool operator==(DateTime const& dt) const { return m_since1601 == dt.m_since1601; }
 
     /// Compare two instances of @DateTime for inequality.
-    bool operator!=(const DateTime& dt) const { return !(*this == dt); }
+    constexpr bool operator!=(const DateTime& dt) const { return !(*this == dt); }
 
     /// Compare the chronological order of two @DateTime instances.
-    bool operator>(const DateTime& dt) const { return this->m_interval > dt.m_interval; }
+    constexpr bool operator>(const DateTime& dt) const
+    {
+      return this->m_since1601 > dt.m_since1601;
+    }
 
     /// Compare the chronological order of two @DateTime instances.
-    bool operator<(const DateTime& dt) const { return this->m_interval < dt.m_interval; }
+    constexpr bool operator<(const DateTime& dt) const
+    {
+      return this->m_since1601 < dt.m_since1601;
+    }
 
     /// Compare the chronological order of two @DateTime instances.
-    bool operator>=(const DateTime& dt) const { return this->m_interval >= dt.m_interval; }
+    constexpr bool operator>=(const DateTime& dt) const
+    {
+      return this->m_since1601 >= dt.m_since1601;
+    }
 
     /// Compare the chronological order of two @DateTime instances.
-    bool operator<=(const DateTime& dt) const { return this->m_interval <= dt.m_interval; }
-
-    /// Create an interval from milliseconds.
-    static IntervalType FromMilliseconds(unsigned int milliseconds)
+    constexpr bool operator<=(const DateTime& dt) const
     {
-      return milliseconds * TicksPerMillisecond;
+      return this->m_since1601 <= dt.m_since1601;
     }
 
-    /// Create an interval from seconds.
-    static IntervalType FromSeconds(unsigned int seconds) { return seconds * TicksPerSecond; }
```

Comment:
What happened to the `Year`/`Month`/`Hour`/etc. constants, so that the user can do `Datetime + 24 * Hour`?